src/discord/discord.py

```diff
@@ -0,0 +1,277 @@
+            permission &= ~int(role_everyone_overrides['deny'])
+            permission |= int(role_everyone_overrides['allow'])
+
+        allow = 0
+        deny = 0
+        for role in user.roles:
+            if role in channel_overwrites:
+                allow |= int(channel_overwrites[role]['allow'])
+                deny |= int(channel_overwrites[role]['deny'])
+        permission &= ~deny
+        permission |= allow
+
+        # check member specific perms
+        if user.user.id in channel_overwrites:
+            # switching of 'deny' and 'allow' intentional
+            permission &= ~int(channel_overwrites[user.user.id]['deny'])
+            permission |= int(channel_overwrites[user.user.id]['allow'])
+
+        if permission & VIEW_CHANNEL == VIEW_CHANNEL:
+            if user not in users_with_access:
+                users_with_access.append(user)
+
+    return users_with_access
+
+async def get_channel(
+    cid: str,
+    id: str = guild_id
+) -> Channel:
+    tok = os.environ.get('TOKEN')
+    url = f'https://discord.com/api/v10/guilds/{id}/channels'
+    result = await _discord_request(url, tok)
+
+    result_json = result.json()
+    channel = [channel for channel in result_json if channel['id'] == cid][0]
+    channel = Channel(channel['id'], channel['type'], channel['guild_id'], channel['name'], channel['permission_overwrites'])
+
+    return channel
+
+async def get_all_channels(
+    id: str = guild_id
+) -> list[str]:
+    tok = os.environ.get('TOKEN')
+    url = f'https://discord.com/api/v10/guilds/{id}/channels'
+    result = await _discord_request(url, tok)
+
+    result_json = result.json()
+    channels = [channel for channel in result_json if channel['type'] != DISCORD_CATEGORY_ID]
+    channel_names = [channel['name'] for channel in channels]
+
+    return channel_names
+
+
+async def get_role_name_by_id(
+    rid: str,
+    id: str = guild_id
+) -> str:
+    roles = await get_all_roles(id)
+    return roles[rid][0]
+
+async def get_role_by_id(
+    rid: str,
+    id: str = guild_id
+) -> dict:
+    tok = os.environ.get('TOKEN')
+    url = f'https://discord.com/api/v10/guilds/{id}/roles'
+    result = await _discord_request(url, tok)
+
+    json_s = result.json()
+    return [role for role in json_s if role['id'] == rid][0]
+
+async def get_user_roles(
+    uid: str,
+    id: str = guild_id
+) -> list[str]:
+    tok = os.environ.get('TOKEN')
+    url = f'https://discord.com/api/v10/guilds/{id}/members/{uid}'
+    result = await _discord_request(url, tok)
+
+    json_s = result.json()
+    return json_s['roles']
+
+async def get_all_roles(
+    id: str = guild_id
+) ->  dict[str, list[str]]:
+    tok = os.environ.get('TOKEN')
+    url = f'https://discord.com/api/v10/guilds/{id}/roles'
+    result = await _discord_request(url, tok)
+
+    json_s = result.json()
+    roles = [([role['id'], [role['name'], role['permissions']]]) for role in json_s]
+    return dict(roles)
+
+async def get_guild_members_with_role(
```

Comment:
I notice you sometimes use the terminology `user` and sometimes `guild_member`. Let's be consistent with terminology, but either is fine with me!

Reply:
I have realized that users and guild_members are different!